.travis.yml

```
@@ -18,8 +18,7 @@ addons:
 
 before_script:
   - pip3 install wheel
-  - pip3 install --upgrade 'meson==0.52.0'
-  - meson --version
+  - pip3 install meson
```

Can we pin this to a specific version like before? i.e.

```suggestion
  - pip3 install --upgrade 'meson==0.54.1'
```

Using meson 0.53 caused Travis to fail without any changes on our part, and took some time to fix - time also potentially wasted assuming our new changes was the culprit.

It'd be best to avoid running into such issues again. I'd prefer updating the version after verifying it doesn't cause any issues.

Likewise, echoing the version info was useful in debugging the issue, so I'd like to keep it.

See 16982f7.

---

Updated, but I'm not sure if I like it that much. The problem in 16982f712633b9f612519a498d5c185fa2611ee2 was that we used a Meson version that wasn't compatible with bionic, it wasn't the fault of Meson. I agree that is annoying when the build fails because of that, but wouldn't then a better option be to just use the Meson version provided by the distro? Bionic has Meson 0.45.1, I don't know if that will still work with gamemode. Also possible would be to go with focal, and take it as a baseline.

---

When I wrote this, Travis didn't have Bionix support yet, it does now but it won't have Focal for a long while: https://docs.travis-ci.com/user/reference/overview/#linux

---

If we're able to use the version of meson packaged with the distro I think that would be ideal. That will still avoid the problem of incompatible versions from pip but will be easier for us to maintain. Although if 0.45.1 ends up being too old, I'm not opposed to sticking to a pinned version from pip.

The main justification is to remove a potential unknown from our tests, regardless of the cause of the problem. I don't want the test environment to change without our knowledge, but always grabbing the latest version via pip has the potential to do exactly that.

---

> When I wrote this, Travis didn't have Bionic support yet, it does now but it won't have Focal for a long while: https://docs.travis-ci.com/user/reference/overview/#linux

I run travis CI instance with focal without any problems. It works, I guess it's just not advertised (yet).